marshal/tinyint/unmarshal.go

```diff
@@ -0,0 +1,73 @@
+package tinyint
```

Comment:
from the package structure it looks a bit weird, marshal tiny int unmarshal, let's change name for top level folder to `serialization`, so whole thing will be `serialization/tinyint/unmarshal.go`

Reply:
in #270 we already have path `marshal/tests/...` , maybe be better do not scatter the code intended for marshal in different folders?

Reply:
I skipped that thing there when I was reviewing that PR, everything from `marshal/tests/` should be moved to `internal/tests` folder.

Reply:
Maybe should still allow users to use `suite` to test their functions?

Reply:
> Maybe should still allow users to use `suite` to test their functions? Please write me what you have chosen.

I don't think we want that, in such case we will need to consider it as an API with all the limitations it brings.

Reply:
So, can you please rename it to serialization and move `marhsal/tests` to `internal/tests`

Reply:
I do this in #278